docs/content/docs/deployment/memory/network_mem_tuning.md

```diff
@@ -0,0 +1,176 @@
+long checkpoint time (for aligned checkpoint). Also, larger buffer sizes
+do not make sense with small `execution.buffer-timeout` because in this case all (or almost all)
+flushed buffers would be sent only partially filled which means that the allocated memory won't be utilized efficiently.
+
+### Selecting the buffer count
+The number of buffers is configured by `taskmanager.network.memory.buffers-per-channel`, `taskmanager.network.memory.floating-buffers-per-gate`.
+For best throughput in most cases we recommend to stick with the default values for the number of exclusive
+and floating buffers. If the amount of in-flight data is causing problems enabling
+[buffer debloating]({{< ref "docs/deployment/memory/network_mem_tuning" >}}#buffer-debloat) is again
+the easiest and the recommended way of trying to deal with those problems. However if for some reason
+this doesn't work for you, here are things to consider when tuning the amount of network buffers.
+
+You should adjust the amount of buffers according to your expected throughput (in `bytes/second`).
+Assigning credits and sending buffers takes some time (around two round trip messages between two nodes).
+This latency depends on your network, but for the sake of simplicity in the healthy local networks we can assume it's `1ms`.
+Next we take [the buffer size]({{< ref "docs/deployment/config" >}}#taskmanager-memory-segment-size).
+With those three numbers (buffer round trip time, buffer size and expected throughput) we can calculate
+the number of required buffers to sustain this throughput by using this simple formula:
+```
+number_of_buffers = expected_throughput * buffer_roundtrip / buffer_size
+```
+For example, with expected throughput of `320MB/s`, round trip latency of `1ms` and the default memory segment size:
+```
+number_of_buffers = 320MB/s * 1ms / 32KB = 10
+```
+This is the number of actively used buffers that we need to achieve the expected throughput with the given round trip latency.
+
+The purpose of floating buffers is to handle the data skew scenario. Ideally number of floating buffers
+(default 8) plus the exclusive buffers (default 2) that belong to that channel should be able to saturate
+the network throughput. However sometimes this is either not feasible or even not necessary.
+Keep in mind that it happens very rarely that only a single channel among all subtasks on your task manager is being used.
+
+The exclusive buffers on the other hand have a purpose to provide a fluent throughput.
+While one buffer is in transit, the other is being filled up. However with larger setups,
+the number of exclusive buffers is the dominant factor defining the amount of in-flight data Flink is using.
+So in case of back pressure in low throughput setups, it might be a good idea to reduce the number of [exclusive buffers]({{< ref "docs/deployment/config" >}}#taskmanager-network-memory-buffers-per-channel).
```

Comment:
```suggestion
## The number of in-flight buffers 

The default settings for exclusive buffers and floating buffers should be sufficient for the maximum throughput.  If the minimum of in-flight data needs to be set, the exclusive buffers can be set to `0` and the memory segment size can be decreased.

### Selecting the buffer size

The buffer collects records in order to optimize network overhead when sending the data portion to the next subtask. The next subtask should receive all parts of the record before consuming it. 

If the buffer size is too small (i.e. less than one record), this can lead to low throughput since the overhead is still pretty large.  

If the buffer size is too large, this can lead to: 
- high memory usage
- huge checkpoint data (for unaligned checkpoints)
- long checkpoint time (for aligned checkpoints)
- inefficient use of allocated memory with a small `execution.buffer-timeout` because flushed buffers would only be sent partially filled 

### Selecting the buffer count

The number of buffers is configured by the `taskmanager.network.memory.buffers-per-channel` and `taskmanager.network.memory.floating-buffers-per-gate` settings. 

For best throughput, we recommend using the default values for the number of exclusive
and floating buffers. If the amount of in-flight data is causing issues, enabling
[buffer debloating]({{< ref "docs/deployment/memory/network_mem_tuning" >}}#buffer-debloat) is recommended. 

You can tune the number of network buffers manually, but consider the following: 

1. You should adjust the number of buffers according to your expected throughput (in `bytes/second`).
Assigning credits and sending buffers takes some time (around two roundtrip messages between two nodes). The latency also depends on your network.

Using the buffer roundtrip time (around `1ms` in a healthy local network), [the buffer size]({{< ref "docs/deployment/config" >}}#taskmanager-memory-segment-size), and the expected throughput, you can calculate the number of buffers required to sustain the throughput by using this formula:
```
number_of_buffers = expected_throughput * buffer_roundtrip / buffer_size
```

For example, with an expected throughput of `320MB/s`, roundtrip latency of `1ms`, and the default memory segment size, 10 is the number of actively used buffers needed to achieve the expected throughput:
```
number_of_buffers = 320MB/s * 1ms / 32KB = 10
```


2. The purpose of floating buffers is to handle data skew scenarios. Ideally, the number of floating buffers (default: 8) and the exclusive buffers (default: 2) that belong to that channel should be able to saturate the network throughput. But this is not always feasible or necessary. It is very rare that only a single channel among all the subtasks in the task manager is being used.

3. The purpose of exclusive buffers is to provide a fluent throughput. While one buffer is in transit, the other is being filled up. With high throughput setups, the number of exclusive buffers is the main factor that defines the amount of in-flight data Flink uses.

In the case of backpressure in low throughput setups, you should consider reducing the number of [exclusive buffers]({{< ref "docs/deployment/config" >}}#taskmanager-network-memory-buffers-per-channel).
```